core/src/actors/session/messages.rs

```diff
@@ -9,3 +10,10 @@ pub struct GetPeers;
 impl Message for GetPeers {
     type Result = SessionUnitResult;
 }
+
+/// Message to announce new inventory items through the network
+#[derive(Clone, Message)]
```

Comment:
why is the clone needed? :wink:

Reply:
The `Clone` is needed for the `Broadcast` method, because this message will be used by #173 to announce a new block to all the outbound peers.

This means that each session will construct an `Inv` message from the `Vec<InvVector>`, which is actually unnecessary because all those messages are exactly the same.

Therefore, the `Broadcast` method could be optimized by constructing the raw message before sending a message to each session actor, although this would require some refactoring.

Reply:
Agreed, would you be ok with creating an issue for that optimisation? I think it would be nice to have it in the future for network messages that are exactly the same for all sessions.

Reply:
Discussed in #198.